libs/state/src/lib/core/operators/distinctUntilSomeChanged.ts

```diff
@@ -0,0 +1,173 @@
+import { MonoTypeOperatorFunction } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+
+/**
+ * @internal
+ */
+export type CompareFn<T> = (x: T, y: T) => boolean;
+/**
+ * @internal
+ */
+export type KeyCompareMap<T extends object> = {
+  [K in keyof T]: CompareFn<T[K]>;
+};
+
+/**
+ * @description
+ *
+ * Returns an Observable that emits all items emitted by the source Observable that are distinct by comparison from
+ * the previous item. You can provide a custom comparison for each key individually by setting a `KeyCompareMap<T>`.
+ * If no comparison is provided for a specified key, an equality check is used by default.
+ *
+ * If properties of the source change, which are not specified for comparison, no change will be emitted.
```

Comment:
I did not see a test for that behavior

Reply:
see the test named `should ignore changes of other keys`